apps/settings/views.py

```diff
@@ -90,9 +91,31 @@ def integrations(request):
     return render(request, "settings/integrations.html", context=data)
 
 
+@csrf_exempt
 def data_portability(request):
     """ Export bookmarks """
-    return render(request, "settings/dataportability.html")
+    from apps.web.models import Bookmarks
+    import csv
+
+    user = request.user
+    bookmarks = Bookmarks.objects.filter(user=user)\
+        .values('title', 'link', 'description')
```

Comment:
I realized that we should probably include more data, so I did this:
```
bookmarks = Bookmarks.objects.filter(user=user)\
            .values('bm_id', 'title', 'link', 'image', 'description', 'read', 'created', 'tags__name', 'body')\
            .order_by('-id')
```
But it returned duplicated items, which I later found out that it was because of the many2many field on the attribute `tags`.

Came up with the following code:
```
def data_portability(request):
    """ Export bookmarks """

    if request.method == "POST":
        user = request.user
        bookmarks = Bookmarks.objects.filter(user=user)\
            .values('bm_id', 'title', 'link', 'image', 'description', 'read', 'created', 'tags__name', 'body')\
            .order_by('-id')

        merged = {}
        if bookmarks:
            # The bookmarks object will contain duplicates because of the many2many field on tags.
            # Items needs to be merged.
            for item in bookmarks:
                if not item["bm_id"] in merged.keys():
                    merged[item["bm_id"]] = item
                else:
                    tags = item["tags__name"]
                    x = merged[item["bm_id"]]
                    if tags and x["tags__name"]:
                        x["tags__name"] += f", {tags}"
                        merged[item["bm_id"]] = x

        response = HttpResponse(content_type='text/csv')
        response['Content-Disposition'] =\
            'attachment; filename="bookie_output.csv"'
        writer = csv.writer(response, delimiter=';')
        writer.writerow(['Title', 'Link', 'Image', 'Description', 'Read', 'Created', 'Tags', 'Body'])
        for key, bm in merged.items():
            writer.writerow([bm.get('title'),
                             bm.get('link'),
                             bm.get('image'),
                             bm.get('description'),
                             bm.get('read'),
                             bm.get('created'),
                             bm.get('tags__name'),
                             bm.get('body')])
        return response
    return render(request, "settings/dataportability.html")
```

The only changes I made was
* to move all the code inside the `if request.method == "POST":` so we don't have to fetch all the items as soon as you visit the page.
* Add more fields to the csv
* Added `;` as delimiter, worked better in excel 😄 
* Merge the same items so we can avoid duplicates

This also means that if a user does not have any bookmarks, the user will simply download an empty file, which I think is okay actually.

If you think it looks good you can update with a new commit and we can merge your PR into master!

Reply:
Hi @mjdubell , your suggestions are awesome! pushing the PR now